src/database/db.js

```diff
@@ -69,24 +110,71 @@ export function getShabadFromID(shabadID, length, language) {
                     : translationJson[lang][field];
                 const translit =
                   getTranslitText(row.Transliterations, language) || defaultTranslation;
-                const englishTranslations = getTranslation("en", "bdb");
-                const punjabiTranslations = getTranslation("pu", "bdb");
-                const spanishTranslations = getTranslation("es", "sn");
+                English = getTranslation("en", "bdb");
+                Punjabi = getTranslation("pu", "bdb");
+                Spanish = getTranslation("es", "sn");
+
+                if (isParagraphMode) {
+                  const isParagraphChange = prevParagraph !== Paragraph;
+                  const isLastIteration = i === len - 1;
+
+                  if (isParagraphChange) {
+                    if (i !== 0) {
+                      paragraphResults.push({
+                        id: `${paragraphID}`,
```

Comment:
We are using paragraphID variable here, but it has not been intialized with any value. The value is initialized below this code block. Correct me if I understood it wrong.

Reply:
Yes, veer ji because it comes with condition so it won't be assigning null/ undefined value.